project/Settings.scala

```diff
@@ -422,9 +422,20 @@ object Settings {
     Test / unmanagedSourceDirectories ++= {
       val testsRootDir = baseDirectory.value.getParentFile.getParentFile
       val sharedTestDir = testsRootDir / "shared/src/test"
+      val scalaVersionDir = CrossVersion
+        .partialVersion(scalaVersion.value)
+        .collect {
+          case (3, _)     => "scala3"
+          case (2, minor) => s"scala2.$minor"
+        }
+        .getOrElse(sys.error("Unsupported Scala version"))
       // Java 8 is reference so start at 9
-      (9 to (Global / javaVersion).value).map { v =>
-        sharedTestDir / s"require-jdk$v"
+      (9 to (Global / javaVersion).value).flatMap { v =>
+        val jdkVersion = s"jdk$v"
+        Seq(
+          sharedTestDir / s"require-$jdkVersion",
+          sharedTestDir / s"require-$jdkVersion-$scalaVersionDir"
```

Comment:
After your comment in the upstream PR I think that `scala-3-require-jdk9` might be better suited. If it's not a problem you can use it instead of the current `require-jdk9-scala3``?

Reply:
Will change, after CI passes.  

I had originally gone with `scala-3-require-jdk9` and then flopped to the other variant.

At some point in the future, I would like to experiment with putting the require-jdk9
under the scala-3, but not today. Getting a reasonable first approximation up and running
if more important.